lib/cosmos/packets/packet.rb

```diff
@@ -218,11 +218,24 @@ def identified?
     # @param id_value [Object] Set to something other than nil to indicate that
     #   this item should be used to identify a buffer as this packet. The
     #   id_value should make sense according to the data_type.
+    # @return [PacketItem] The new packet item
     def define_item(name, bit_offset, bit_size, data_type, array_size = nil, endianness = @default_endianness, overflow = :ERROR, format_string = nil, read_conversion = nil, write_conversion = nil, id_value = nil)
       item = super(name, bit_offset, bit_size, data_type, array_size, endianness, overflow)
       packet_define_item(item, format_string, read_conversion, write_conversion, id_value)
     end
 
+    # Add an item to the packet by adding it to the items hash. It also
+    # resizes the buffer to accomodate the new item.
+    #
+    # @param item [PacketItem] Item to add to the packet
+    # @return [PacketItem] The same packet item
+    def define(item)
+      item = super(item)
+      update_id_items(item)
+      update_limits_items_cache()
```

Comment:
I initially forgot this but created a test case for it once I had something else not work. I don't have an append method in packet because Structure's append calls define which calls this method.